webapp/src/main/java/com/box/l10n/mojito/rest/openai/OpenAIConfig.java

```diff
@@ -0,0 +1,56 @@
+package com.box.l10n.mojito.rest.openai;
+
+import com.box.l10n.mojito.openai.OpenAIClient;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConfigurationProperties(prefix = "l10n.openai")
+public class OpenAIConfig {
+
+  boolean enabled;
+
+  String apiKey;
+
+  String host;
+
+  Map<String, String> customHeaders = new HashMap<>();
+
+  List<String> allowRestrictedHeaders;
+
+  public void setApiKey(String apiKey) {
+    this.apiKey = apiKey;
+  }
+
+  public void setCustomHeaders(Map<String, String> customHeaders) {
+    this.customHeaders = customHeaders;
+  }
+
+  public void setEnabled(boolean enabled) {
+    this.enabled = enabled;
+  }
+
+  public void setHost(String host) {
+    this.host = host;
+  }
+
+  public void setAllowRestrictedHeaders(List<String> allowRestrictedHeaders) {
+    this.allowRestrictedHeaders = allowRestrictedHeaders;
+  }
+
+  @Bean
+  @ConditionalOnProperty(value = "l10n.openai.enabled", havingValue = "true")
+  public OpenAIClient openAIClient() {
+    if (!allowRestrictedHeaders.isEmpty()) {
```

Comment:
Is this required to be able to set a single `Host` header? It might help to add a small note re that behaviour.

Reply:
It can be used to allow a set of header modifications that are restricted by default in the jdk. In our case it was the host header that was causing an issue, but if needed it will unrestrict the list of header names passed in